### PR TITLE
F/process data

### PR DESCRIPTION
**Pull request by Gnwin**
## Blog App2 : f/process-data

In this milestone, 
- ruby on rails was setup with postgres
- project schema was built
- models were created
- models were used to insert sample data with rails console

example:
Screenshot 2022-07-23 at 2 26 56 PM

All requirements have been met according to this document:
https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -2,4 +2,12 @@ class Post < ApplicationRecord
   has_many :comment
   has_many :like
   belongs_to :user
+
+  def increment_user_posts_count
```
Comment (medaminedev66) [inline app/models/post.rb:10]: [OPTIONAL]  I suggest using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #increment_user_posts_count method every time a new post has been saved to the database, then make the increment_user_posts_count method private.

File: app/models/like.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class Like < ApplicationRecord
   belongs_to :user
   belongs_to :post
+
+  def increment_posts_likes_counter
+    post.increment!(:likes_counter)
+  end
```
Comment (medaminedev66) [inline app/models/like.rb:11]: [OPTIONAL] I suggest using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #increment_posts_likes_counter method every time a new like has been saved to the database, then make the increment_posts_likes_counter method private.

File: app/models/comment.rb [hunk 3]
```diff
@@ -1,4 +1,8 @@
 class Comment < ApplicationRecord
   belongs_to :user
   belongs_to :post
+
+  def increment_posts_comments_counter
+    post.increment!(:comments_counter)
+  end
```
Comment (medaminedev66) [inline app/models/comment.rb:11]: [OPTIONAL] As I mentioned in the previous comments, you may use `after_save` to call the #increment_posts_comments_counter method every time a new comment has been saved to the database.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @Gnwin   😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- There are no linter errors
- You have a descriptive PR

## Required Changes ♻️
- [ ] Your readme isn't descriptive, and it doesn't provide enough details on the project. Please update your Readme with the proper one.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @Gnwin 😃,

Great work on making the changes requested in the previous review 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights


 ## Required Changes ♻️
- [ ] As I requested in the previous review, you should update your Readme file so that your repo is professional. Your Readme file is still not descriptive and it doesn't provide enough details on the project. You may use [Microver template](https://github.com/microverseinc/readme-template) as an example.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — APPROVED** [review]
# STATUS: Approved 🟢 ✔️ 

## Hi @Gnwin 👋, 

Well done for implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**